kt/common/src/main/kotlin/godot/common/extensions/StringExtensions.kt

```diff
@@ -48,6 +48,48 @@ fun String.convertToSnakeCase(): String =
         else accumulator.append(character)
     }.toString()
 
+fun String.toUpperSnakeCase(): String {
+    if (this.isEmpty()) return this
+
+    val result = StringBuilder()
```

Comment:
why use `StringBuilder` instead of just `buildString{}`? IMO that's more concise